### PR TITLE
fix: set tls servername

### PR DESCRIPTION
**Pull request by ronag**
Fixes: https://github.com/mcollina/undici/issues/217

### CODE REVIEW

**Comment by ronag** [comment]
Fails on ubuntu for some reason... if someone wants to pick up from here feel free to do so...

Maybe @kmamal?

**Comment by ronag** [comment]
Would probably better to have a local https server to test against instead of github but that's a bit over my head at the moment

**Comment by szmarczak** [comment]
Sorry, I'm unable to take a deeper look as of now. I'm gonna be active again since the 18th due to the maturity exams I'm having. I'll definitely take a look then.

**Comment by ronag** [comment]
> Sorry, I'm unable to take a deeper look as of now. I'm gonna be active again since the 18th due to the maturity exams I'm having. I'll definitely take a look then.

No worries. Good luck with your exams!

File: lib/client-base.js [hunk 1]
```diff
@@ -552,14 +556,37 @@ function resume (client) {
...
+        if (client[kSocket]) {
+          client[kSocket].destroy(new Error('new servername'))
+        }
+      }
+
+      // TODO: secureConnecting?
+      // TODO: authorized?
+      if (client[kSocket] && !client[kSocket].servername) {
+        return
+      }
```
[the remark this answers is not in the document]
Comment (bnoordhuis) [reply lib/client-base.js:589]: Good question. I _think_ it's fine (client can't send data until handshake completes; `'error'` event is emitted on auth failure) but if you want to be sure, please open a nodejs/node issue.

**Pull request merged**